src/blocksparse/block.jl

```diff
@@ -62,6 +62,8 @@ convert(::Type{Block{N}}, t::Tuple) where {N} = Block{N}(t)
 
 (::Type{IntT})(b::Block{1}) where {IntT<:Integer} = IntT(only(b))
 
+getindex(B::Block{N}, R::UnitRange{Int64}) where {N} = Block(B.data[R])
```

Comment:
Where and how is this being used?

Keep in mind that this isn't type stable (Julia can't infer the length of the `Block` afterward), for example:
```julia
julia> t = (1,2,3)
(1, 2, 3)

julia> @code_warntype t[2:3]
Variables
  #self#::Core.Const(getindex)
  t::Tuple{Int64, Int64, Int64}
  r::UnitRange{Int64}
  @_4::Union{Nothing, Tuple{Int64, Int64}}
  @_5::Int64
  @_6::Vector{Int64}
  ri::Int64

Body::Tuple{Vararg{Int64, N} where N}
[...]
```
If you're interested in something like:
```julia
b = Block(1, 2, 3)
b[1:end-1]
b[2:end]
```
there are type-stable alternatives I could suggest like:
```julia
julia> Base.front(t)
(1, 2)

julia> @code_warntype Base.front(t)
Variables
  #self#::Core.Const(Base.front)
  t::Tuple{Int64, Int64, Int64}

Body::Tuple{Int64, Int64}
1 ─      nothing
│   %2 = Core._apply_iterate(Base.iterate, Base._front, t)::Tuple{Int64, Int64}
└──      return %2

julia> Base.tail(t)
(2, 3)

julia> @code_warntype Base.tail(t)
Variables
  #self#::Core.Const(Base.tail)
  x::Tuple{Int64, Int64, Int64}

Body::Tuple{Int64, Int64}
1 ─ %1 = Core._apply_iterate(Base.iterate, Base.argtail, x)::Tuple{Int64, Int64}
└──      return %1
```
Also other functions like this in [TupleTools.jl](https://jutho.github.io/TupleTools.jl/stable/) and type stable indexing into tuples with [StaticNumbers.jl](https://perrutquist.github.io/StaticNumbers.jl/dev/index.html):
```julia
julia> t[@stat 2:3]
(2, 3)

julia> @code_warntype t[@stat 2:3]
Variables
  #self#::Core.Const(getindex)
  t::Tuple{Int64, Int64, Int64}
  r::Core.Const(static(2:3))
  #25::StaticNumbers.var"#25#26"{Tuple{Int64, Int64, Int64}, LengthUnitRange{Int64, StaticInteger{1}, StaticInteger{2}}}

Body::Tuple{Int64, Int64}
1 ─      nothing
│   %2 = StaticNumbers.:(var"#25#26")::Core.Const(StaticNumbers.var"#25#26")
│   %3 = Core.typeof(t)::Core.Const(Tuple{Int64, Int64, Int64})
│   %4 = Core.typeof(r)::Core.Const(LengthUnitRange{Int64, StaticInteger{1}, StaticInteger{2}})
│   %5 = Core.apply_type(%2, %3, %4)::Core.Const(StaticNumbers.var"#25#26"{Tuple{Int64, Int64, Int64}, LengthUnitRange{Int64, StaticInteger{1}, StaticInteger{2}}})
│        (#25 = %new(%5, t, r))
│   %7 = #25::StaticNumbers.var"#25#26"{Tuple{Int64, Int64, Int64}, LengthUnitRange{Int64, StaticInteger{1}, StaticInteger{2}}}
│   %8 = StaticNumbers.length(r)::Core.Const(static(2))
│   %9 = StaticNumbers.ntuple(%7, %8)::Tuple{Int64, Int64}
└──      return %9
```

Reply:
Good catch. I ended up finding a way to write the code on the ITensor side that doesn't need this `getindex` function at all, so I was able to remove it entirely.